lib/zlib.js

```diff
@@ -170,319 +169,322 @@ function flushCallback(level, strategy, callback) {
 // This thing manages the queue of requests, and returns
 // true or false if there is anything in the queue when
 // you call the .write() method.
-class Zlib extends Transform {
-  constructor(opts, mode) {
-    opts = opts || {};
-    super(opts);
+function Zlib(opts, mode) {
+  if (!(this instanceof Zlib))
```

Comment:
I don't think this is needed since it's not explicitly exported and is used only as a base "class." Removing it will also help performance-wise.